lib/module.js

```diff
@@ -19,25 +23,40 @@ module.exports = function (moduleOptions) {
     ...moduleOptions
   }
 
-  if (options.copyExistingFiles) {
-    copyExistingNetlifyFiles.bind(this)(options)
-  }
+  this.nuxt.hook('generate:done', () => {
+    if (options.netlifyToml) {
+      programmaticallyCreateToml.bind(this)(options)
+      // prevent netlify.toml from being copied also
+      filesToCopy = filesToCopy.filter(file => file !== TOML_FILE_NAME)
+    }
+
+    if (options.copyExistingFiles) {
+      copyExistingNetlifyFiles.bind(this)(options)
+    }
+  })
+}
+
+function programmaticallyCreateToml ({ netlifyToml }) {
+  let tomlObj = netlifyToml
+  if (typeof tomlObj === 'function') { tomlObj = netlifyToml() }
+  if (typeof tomlObj !== 'object') { throw new TypeError('"netlifyToml" must be an object, or a function that returns an object') }
+  const toml = tomlStringify(tomlObj)
+  const destination = resolve(this.options.rootDir, this.options.generate.dir, TOML_FILE_NAME)
+  writeFileSync(destination, toml)
 }
 
 function copyExistingNetlifyFiles ({ existingFilesDirectory }) {
```

Comment:
Maybe we could pass the file names to copy as an argument?

Reply:
sure, will do in the next days 👍